opshin/tests/test_Unions.py

```diff
@@ -0,0 +1,242 @@
+import unittest
+
+import hypothesis
+from hypothesis import given
+from hypothesis import strategies as st
+from .utils import eval_uplc_value
+from . import PLUTUS_VM_PROFILE
+from opshin.util import CompilerError
+
+hypothesis.settings.load_profile(PLUTUS_VM_PROFILE)
+
+from .test_misc import A
+
+
+def to_int(x):
+    if isinstance(x, A):
+        return 5
+    elif isinstance(x, int):
+        return 6
+    elif isinstance(x, bytes):
+        return 7
+    elif isinstance(x, list):
+        return 8
+    elif isinstance(x, dict):
+        return 9
+    return False
+
+
+union_types = st.sampled_from([A(0), 10, b"foo", [1, 2, 3, 4, 5], {1: 2, 2: 3}])
+
+
+class Union_tests(unittest.TestCase):
```

Comment:
Please add a testcase that shows that Unioning PlutusData of the same constructors is illegal. Also one for str, bool (the non-allowed primitive type).